spec/migrations/20220809152005_fix_ibm_power_vc_sti_classes_spec.rb

```diff
@@ -0,0 +1,111 @@
+  let(:floating_ip_stub)           { migration_stub(:FloatingIp) }
+  let(:network_port_stub)          { migration_stub(:NetworkPort) }
+  let(:network_router_stub)        { migration_stub(:NetworkRouter) }
+  let(:security_group_stub)        { migration_stub(:SecurityGroup) }
+  let(:cloud_volume_stub)          { migration_stub(:CloudVolume) }
+  let(:cloud_volume_backup_stub)   { migration_stub(:CloudVolumeBackup) }
+  let(:cloud_volume_snapshot_stub) { migration_stub(:CloudVolumeSnapshot) }
+  let(:cloud_volume_type_stub)     { migration_stub(:CloudVolumeType) }
+
+  migration_context :up do
+    it "Fixes IBM PowerVC Network STI classes" do
+      ibm_power_vc_network_klass = "ManageIQ::Providers::IbmPowerVc::NetworkManager"
+      openstack_network_klass = "ManageIQ::Providers::Openstack::NetworkManager"
+
+      network_manager = ems_stub.create(:type => ibm_power_vc_network_klass)
+
+      cloud_network         = cloud_network_stub.create(:ext_management_system => network_manager, :type => "#{openstack_network_klass}::CloudNetwork")
+      cloud_network_public  = cloud_network_stub.create(:ext_management_system => network_manager, :type => "#{openstack_network_klass}::CloudNetwork::Public")
+      cloud_network_private = cloud_network_stub.create(:ext_management_system => network_manager, :type => "#{openstack_network_klass}::CloudNetwork::Private")
+      cloud_subnet          = cloud_subnet_stub.create(:ext_management_system => network_manager)
+      floating_ip           = floating_ip_stub.create(:ext_management_system => network_manager)
+      network_port          = network_port_stub.create(:ext_management_system => network_manager)
+      network_router        = network_router_stub.create(:ext_management_system => network_manager)
+      security_group        = security_group_stub.create(:ext_management_system => network_manager)
+
+      migrate
+
+      expect(cloud_network.reload.type).to         eq("#{ibm_power_vc_network_klass}::CloudNetwork")
+      expect(cloud_network_public.reload.type).to  eq("#{ibm_power_vc_network_klass}::CloudNetwork::Public")
+      expect(cloud_network_private.reload.type).to eq("#{ibm_power_vc_network_klass}::CloudNetwork::Private")
+      expect(cloud_subnet.reload.type).to          eq("#{ibm_power_vc_network_klass}::CloudSubnet")
+      expect(floating_ip.reload.type).to           eq("#{ibm_power_vc_network_klass}::FloatingIp")
+      expect(network_port.reload.type).to          eq("#{ibm_power_vc_network_klass}::NetworkPort")
+      expect(network_router.reload.type).to        eq("#{ibm_power_vc_network_klass}::NetworkRouter")
+      expect(security_group.reload.type).to        eq("#{ibm_power_vc_network_klass}::SecurityGroup")
+    end
+
+    it "Fixes IBM PowerVC Storage (Cinder) STI classes" do
+      ibm_power_vc_cinder_klass = "ManageIQ::Providers::IbmPowerVc::StorageManager::CinderManager"
+      openstack_cinder_klass = "ManageIQ::Providers::Openstack::StorageManager::CinderManager"
```

Comment:
Were you going to use this to set the "wrong" types on the records?

Reply:
oops, copy pasted both of these when only the ibm_power_vc is needed